src/test/java/edu/kis/powp/jobs2d/gui/extensions/ModifierListener.java

```diff
@@ -0,0 +1,39 @@
+package edu.kis.powp.jobs2d.gui.extensions;
+
+import edu.kis.powp.jobs2d.features.driverTransofrmation.TransformationModifier;
+import edu.kis.powp.jobs2d.features.driverTransofrmation.TransformingDriver;
+
+import java.awt.event.ActionEvent;
+import java.awt.event.ActionListener;
+
+public class ModifierListener implements ActionListener
+{
+    private boolean enabled;
+    private TransformationModifier modifier;
+    private TransformingDriver lineTrans, specialLineTrans;
+
+    public ModifierListener(TransformationModifier modifier, TransformingDriver lineTrans, TransformingDriver specialLineTrans)
```

Comment:
There are at least two disadvantages of this solution:
- Fixing the number of supported TransformingDrivers (i.e., two) is no OCP. If another driver is added to drivers it will not be automatically supported by modifiers, and it would require changes here.
- If a new driver would not be added as a TransformingDriver modifiers would not work.

Adding a new driver to application (and drivers menu) is done using addDriver method. I think you should think about changing the implementation of addDriver, so it would be responsible for:
- decorating a new driver with TransformingDriver
- adding the driver to ModifierListener for management of transforming extensions.

Reply:
TBD